internal/controller/clientprofile_controller.go

```diff
@@ -130,6 +130,10 @@ func (r *ClientProfileReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 func (r *ClientProfileReconcile) reconcile() error {
 	if err := r.loadAndValidate(); err != nil {
+		if client.IgnoreNotFound(err) == nil {
```

Comment:
I think what you want is to compare to a Not Found error, not to negate a negation of a Not Found error
```suggestion
		if k8serrors.IsNotFound(err) {
```

Reply:
done.